In [ ]:
install.packages("tidyverse")
library(tidyverse)
set.seed(42)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [ ]:
clinical_data_regression <- tibble(
  patient_ID = 1:100,
  age = sample(20:80, 100, replace = TRUE) ,
  weight = sample(50:110, 100, replace = TRUE) ,
  bp_observed = 90 + (0.4 * age) + (0.3 * weight) + rnorm(100, mean = 0, sd = 5)
)
head(clinical_data_regression)

patient_ID,age,weight,bp_observed
<int>,<int>,<int>,<dbl>
1,68,73,135.7661
2,56,98,142.3276
3,20,67,115.9887
4,44,54,123.1882
5,29,95,131.0410
6,55,89,139.2958


In [ ]:
# Building The Linear Regression Model
model <- lm(bp_observed ~ age + weight, data = clinical_data_regression)
summary(model)


Call:
lm(formula = bp_observed ~ age + weight, data = clinical_data_regression)

Residuals:
    Min      1Q  Median      3Q     Max 
-9.5315 -2.3761  0.1578  2.7734 14.1221 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 89.98911    2.87491   31.30   <2e-16 ***
age          0.40480    0.02792   14.50   <2e-16 ***
weight       0.29107    0.02750   10.58   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 4.633 on 97 degrees of freedom
Multiple R-squared:  0.742,	Adjusted R-squared:  0.7367 
F-statistic: 139.5 on 2 and 97 DF,  p-value: < 2.2e-16


# Model Interpretation:
### This Linear Regression Model Predicts The Observed Blood Pressure Based On Two Factors: Age And Weight.

Call:
lm(formula = bp_observed ~ age + weight, data = clinical_data_regression)
### This Line Shows The Equation Used. The Model Predicts Blood Pressure Using Two Factors: Age And Weight.


Residuals:
    Min      1Q  Median      3Q     Max
-9.5315 -2.3761  0.1578  2.7734 14.1221
### These Numbers Show The Difference Between The Real Data And What The Model Predicted (The Prediction Error) .
####1. Min & Max:The Largest Underestimation Was Around -9.5, And The Largest Overestimation Was Around 14.1.
####2. 1Q , Median , 3Q: The Median Error Is Close To 0 (0.15), Which Means The Model's Presictions Are Well-balanced And Accurate On Average. 1Q (First Quartile), 25% Of The Prediction Errors Are Below -2.37. This Means For A Quarter Of The Data, The Prediction Was A Bit Too High Compared To The Real BP. 3Q (Third Quartile), 75% Of The Prediction Errors Are Below 2.7 (Or 25% Are Above It). This Means For Another Quarter Of The Data, The Prediction Was A Bit Too Low.
##### Summary Of 1Q, Median, And 3Q Together: These Values Show How The Errors Are Distributed. Because -2.37 And 2.77 Are Relatively Close To 0 And Evenly Balanced On Both Sides, It Proves That The Model Does Not Have A Huge Bias Toward Overpredicting Or Underpredicting.



Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 89.98911    2.87491   31.30   <2e-16 ***
age          0.40480    0.02792   14.50   <2e-16 ***
weight       0.29107    0.02750   10.58   <2e-16 ***

### This Table Shows How Much Each Factor Affects Blood Pressure.
#### Estimate: The Calculated Effect Size Or Weight Of Each Factor.
#### Intercept = 89.98911: The Baseline Blood Pressure. If A Person's Age Is 0 And Wight Is 0, Their Predicted Blood Pressure Is About 90.
#### Age = 0.40480 / Weight = 0.29107: For Every 1-unit Increase In Age Or Weight, Blood Pressure Increases By The Equal Units.
#### Std.Error: The Standard Error. Smaller Numbers Mean Higher Precision.
#### t-value: A Test Score. Higher Values (Far From 0) Mean The Factor Is Highly Important To The Model.
#### Pr(>|t|) Or p-Value: The Probability That The Results Happened By Pure Chance.<2e-16 ==> The p-value Is Almost 0, Which Means That Both Factors Are Highly Statistically Significant.
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


Residual standard error: 4.633 on 97 degrees of freedom
#### On Average, The Model's Predictions Are Off By 4.633 Units Of BP. The Data Has 97 Degrees Of Freedom (Based On Sample Size).
Multiple R-squared:  0.742,	Adjusted R-squared:  0.7367
#### 74.2% Of The Changes In BP Are Explained By Age And Weight (This Is A Very Strong And Good Fit). A Modified Version Of R-Squared That Adjusts For The Number Of Factors In The Model (Still Very Strong At 73.6%).
F-statistic: 139.5 on 2 and 97 DF,  p-value: < 2.2e-16
#### Tests If The Whole Model Is Useful. Since The P-Value Is Extremely Small, The Overall Model Is Highly Significant And Perfectly Valid.